encode_test.go

```diff
@@ -67,228 +66,754 @@ func TestEncoder_EncodeResponse(t *testing.T) {
-		e          *Encoder
-		args       args
-		err        error
-		statusFunc func(e *Encoder, err error) error
-		wantStatus int
+		name         string
+		e            *Encoder
+		args         args
+		encodeMethod func(e *Encoder, msg string) error
+		setupEncoder func(e *MockHTTPEncoder, w http.ResponseWriter) HTTPEncoder
+		wantErr      bool
+		wantStatus   int
 	}{
 		{
-			name: "Successfully writes unauthorized",
+			name: "BadRequestMessage()",
 			args: args{
-				err: err,
-			},
-			err: err,
-			statusFunc: func(e *Encoder, err error) error {
-				return e.Unauthorized(err)
+				message: "Testing",
+				err:     nil,
 			},
-			wantStatus: http.StatusUnauthorized,
-		},
-		{
-			name: "Successfully writes bad request",
-			args: args{
-				err: err,
+			setupEncoder: func(e *MockHTTPEncoder, _ http.ResponseWriter) HTTPEncoder {
+				e.EXPECT().Encode(&MessageResponse{Message: "Testing"}).Return(nil).AnyTimes()
+				return e
 			},
-			err: err,
-			statusFunc: func(e *Encoder, err error) error {
-				return e.BadRequest(err)
+			encodeMethod: func(e *Encoder, msg string) error {
+				return e.BadRequestMessage(context.Background(), msg)
 			},
 			wantStatus: http.StatusBadRequest,
+			wantErr:    true,
 		},
 		{
-			name: "Successfully writes internal server error",
+			name: "BadRequestMessage() with error",
 			args: args{
-				err: err,
-			},
-			err: err,
-			statusFunc: func(e *Encoder, err error) error {
-				return e.InternalServerError(err)
+				message: "Testing",
+				err:     nil,
 			},
-			wantStatus: http.StatusInternalServerError,
-		},
-		{
-			name: "Successfully writes NotFound",
-			args: args{
-				err: err,
+			setupEncoder: func(e *MockHTTPEncoder, _ http.ResponseWriter) HTTPEncoder {
+				e.EXPECT().Encode(&MessageResponse{Message: "Testing"}).Return(errors.New("big error")).AnyTimes()
+				return e
 			},
-			err: err,
-			statusFunc: func(e *Encoder, err error) error {
-				return e.NotFound(err)
+			encodeMethod: func(e *Encoder, msg string) error {
+				return e.BadRequestMessage(context.Background(), msg)
 			},
-			wantStatus: http.StatusNotFound,
+			wantStatus: http.StatusBadRequest,
+			wantErr:    true,
 		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			e := NewMockHTTPEncoder(ctrl)
+			recorder := httptest.NewRecorder()
+			encoder := &Encoder{
+				encoder: tt.setupEncoder(e, recorder),
+				w:       recorder,
+			}
+			if err := tt.encodeMethod(encoder, tt.args.message); (err != nil) != tt.wantErr {
+				t.Errorf("Encoder.Method() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if recorder.Result().StatusCode != tt.wantStatus {
+				t.Errorf("Encoder.Method() wanted status code %d, got %d", tt.wantStatus, recorder.Result().StatusCode)
+			}
+		})
+	}
+}
+
+func TestEncoder_withBody(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		statusCode int
+		message    string
+		err        error
```

Comment:
statusCode and err are unused

Reply:
I believe "err" should be removed.

However "statusCode" should remain and `tt.encodeMethod(encoder, tt.wantStatus, tt.args.message)` should be replaced with `tt.encodeMethod(encoder, tt.args.statusCode, tt.args.message)`

Reply:
Done